source/secure_tunneling.c

```diff
@@ -2575,25 +2637,21 @@ int aws_secure_tunnel_stream_start(
         return AWS_ERROR_IOTDEVICE_SECURE_TUNNELING_INCORRECT_MODE;
     }
 
-    uint8_t message_protocol_version = s_aws_secure_tunnel_message_min_protocol_check(message_options);
-    if (secure_tunnel->connections->protocol_version != 0 &&
-        message_protocol_version != secure_tunnel->connections->protocol_version) {
+    if (!s_aws_secure_tunnel_protocol_version_match_check(secure_tunnel, message_options)) {
```

Comment:
Is it possible for `secure_tunnel->connections->protocol_version` to be 0 at this point? e.g. if a source device connects to a secure tunnel service, and the first action it does is attempt to send a stream start, isn't its protocol_version currently 0? If it is, then this will always fail as the message will always be > 0.

Reply:
I removed it by mistake, the check for 0 should be performed